protostar/cheatable_starknet/cheatable_cached_state.py

```diff
@@ -48,6 +47,18 @@ def __init__(
         self.contract_address_to_block_timestamp: dict[Address, int] = {}
         self.contract_address_to_block_number: dict[Address, int] = {}
 
+    def get_pranked_address(self, target_address: Address) -> Optional[Address]:
+        if target_address in self._target_address_to_pranked_address:
+            return self._target_address_to_pranked_address[target_address]
+        return None
+
+    def set_pranked_address(self, target_address: Address, pranked_address: Address):
+        self._target_address_to_pranked_address[target_address] = pranked_address
+
+    def remove_pranked_address(self, target_address: Address):
+        if target_address in self._target_address_to_pranked_address:
+            del self._target_address_to_pranked_address[target_address]
+
```

Comment:
These methods hide implementation details. Data can be stored in a dict, Defaultdict or in any other form and Cheater doesn't have to know that.

Reply:
Fair point. Although i'm afraid this will bloat the class in the long run.

Reply:
From what i've noticed - putting methods in the cheatable state encourage other developers to put more logic in that class.

Reply:
Do as you wish though, explanation convinced me on the technical level.

Reply:
Then it can be sliced per functionality. I'd rename the `Cheater` abstract class to `CheatcodeState`. Then, those methods could be moved to `PrankCheatcodeState` together with the copy and apply boilerplate.